addons/mail/models/mail_activity.py

```diff
@@ -356,7 +356,8 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
             self._cr.execute("""
                 SELECT DISTINCT activity.id, activity.res_model, activity.res_id
                 FROM "%s" activity
-                WHERE activity.id = ANY (%%(ids)s)""" % self._table, dict(ids=list(sub_ids)))
+                WHERE activity.id = ANY (%%(ids)s)
+                AND activity.res_id != 0""" % self._table, dict(ids=list(sub_ids)))
```

Comment:
How could this happen ? This is a required field, and always used through a document owning the activity.